portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -19,14 +19,145 @@
+import com.google.appengine.api.datastore.Query.SortDirection;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+/** 
+ * The DataServlet is the servlet which handles GET and POST requests to display 
+ * and update comments.
+ */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-  @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
-  }
+    private static int numberOfShowedComments = 3;    
+
+    /**
+     * Queries the entities in Datastore and sends the response in JSON format
+     */
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        ArrayList<Comment> comments = new ArrayList<>();
+        int index = 0;
+
+        Query query = new Query("Comment");
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        PreparedQuery results = datastore.prepare(query);
+        for (Entity entity : results.asIterable()) {
+            if(index < numberOfShowedComments){
+                index++;
+            } else{
+                break;
+            }
+            String name = (String) entity.getProperty("name");
+            String content = (String) entity.getProperty("content");
+            comments.add(new Comment(name, content));
+        }
+
+        Gson gson = new Gson();
+        String json = gson.toJson(comments);
+        response.setContentType("application/json");
+        response.getWriter().println(json);
+    }
+
+    /**
+     * Obtain parameters of the HTML form and creates an entity in Datastore.
+     */
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String name = null;
+        String content = null;
+        boolean owo = false;
+        boolean uwu = false;
+
+        try {
+            name = getParameter(request, "name-input", "");
+            content = getParameter(request, "text-input", "");
+            owo = getBooleanParameter(request, "owo", false);
+            uwu = getBooleanParameter(request, "uwu", false);
+            DataServlet.numberOfShowedComments = getIntParameter(request, "num-input", 3);
+        } catch (NumberFormatException nfe) {
+            response.sendError(400, "Comment amount must be a number");
+            return;
+        } catch (Exception e){
+            response.sendError(500, "Error");
+            return;
+        }
+
+        String[] commentWords = content.split(" ", 0);
+        String commentConcatenated = "";
+        for(String word : commentWords){
+            if(owo){
+                word += "owo";
+            } else if (uwu){
+                word += "uwu";
+            } 
+            commentConcatenated += word + " ";
+        }
```

Comment:
This seems like a nice self contained block of code which may be reused and would probably make a nice function

Reply:
Sounds good!